src/backend/distributed/planner/distributed_planner.c

```diff
@@ -952,6 +932,17 @@ CreateDistributedPlan(uint64 planId, Query *originalQuery, Query *query, ParamLi
 			distributedPlan =
 				CreateInsertSelectPlan(planId, originalQuery, plannerRestrictionContext);
 		}
+		else if (InsertSelectIntoLocalTable(originalQuery))
+		{
+			if (hasUnresolvedParams)
```

Comment:
Agreed, though I'm not sure that explanation is still entirely accurate for this specific case. 

One reason is that we recurse into CreateDistributedPlan, which may at that time return NULL because of unresolved parameters. Better to return early in that case.